### ISSUE TITLE
[Distribution Support Request] openSUSE Tumbleweed

### ISSUE THREAD

**Issue by RokeJulianLockhart** [PowerShell/PowerShell#20469]
### Name of the Distribution

cpe:/o:opensuse:tumbleweed

### Version of the Distribution

cpe:/o:opensuse:tumbleweed:20231006

### Package Types

- [ ] Deb
- [X] RPM
- [ ] Tar.gz
- [ ] Snap - Stop! Please file your issue in [PowerShell-Snap](https://github.com/powershell/powershell-snap) instead.

### Processor Architecture

x86-64

### .NET Core Support

- [X] The version and architecture of the Distribution is [supported by .NET Core](https://github.com/dotnet/core/blob/master/release-notes/5.0/5.0-supported-os.md#linux).

### Distribution Requirements

- [X] The version of the Distribution is supported for at least one year.
- [X] The version of the Distribution is not an [interim release](https://ubuntu.com/about/release-cycle) or equivalent.

### Exemption Justification

https://github.com/PowerShell/PowerShell/issues/16446#issuecomment-1136368452 doesn't apply to this, because that was Leap, which is supported by the RedHat `.RPM` package. Tumbleweed, however, uses versions which are newer.

### Progress

- [ ] An issue has been filed to create a Docker image in [PowerShell-Docker](https://github.com/powershell/powershell-docker)

### For PowerShell Team **ONLY**

- [ ] Docker image created
- [ ] Docker image published
- [ ] Distribution tested
- [ ] Update `packages.microsoft.com` deployment
- [ ] [Lifecycle](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/docs-conceptual/PowerShell-Support-Lifecycle.md) updated
- [ ] Documentation Updated

**Comment by rhubarb-geek-nz** [comment]
I would suggest that building packages for rolling releases like tumbleweed are problematic because there is no one tumbleweed, it is continually changing. So the phrase "The version of the Distribution is supported for at least one year." doesn't really apply.

**Comment by RokeJulianLockhart** [comment]
@rhubarb-geek-nz, indeed, but considering that the changes made aren't generally monumental - they're as iterative as any home user would want - it's more comparative to a very conservative ring of Windows Insider, like Beta (or even Release Candidate in some cases, since Windows is now fairly modular after its MSIX-ification).

**Comment by rhubarb-geek-nz** [comment]
>nobody would ever beat you up over a version breaking.

I don't recommend that approach, people may be using the releases downloads to directly create docker images in automated systems.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/20469#issuecomment-1889958997

It also doesn't support being set as the login shell, per https://github.com/PowerShell/PowerShell-Snap/issues/101#issue-2082772353.

**Comment by rhubarb-geek-nz** [comment]
See [chsh -s /snap/bin/powershell: powershell is an invalid shell](https://github.com/orgs/PowerShell/discussions/20962)

My understanding is that snap packages are self contained hence cannot affect system files such as /etc/shells when installed.

Whereas RPM and Debian packages can update that in an automatic installation script

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/20469#issuecomment-1892859391

@rhubarb-geek-nz, please see https://snapcraft.io/docs/classic-confinement#:~:text=primarily%2C%20this%20level%20of%20confinement%20is%20used%20for%20applications%20that%20need%20access%20to%20arbitrary%20binaries%20on%20the%20system%2C%20which%20is%20not%20possible%20when%20using%20strict%20confinement. I can't tell you whether it allows it to access what we discuss, but I see no reason why it would not.

**Comment by rhubarb-geek-nz** [comment]
Yes, I see nothing about snap packages having installation scripts that can update the /etc directory, specifically the /etc/shells file. If you manually append the path to the executable to that file, can you then use it as a shell?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/20469#issuecomment-1892881079

I don't know. After all, just adding it to `/etc/shells` doesn't mean that it's immediately the login shell, right?

<blockQuote>

```.txt
/bin/ash
/bin/bash
/bin/csh
/bin/dash
/bin/false
/bin/ksh
/bin/ksh93
/bin/mksh
/bin/pdksh
/bin/sh
/bin/tcsh
/bin/true
/bin/zsh
/usr/bin/csh
/usr/bin/dash
/usr/bin/ksh
/usr/bin/ksh93
/usr/bin/mksh
/usr/bin/passwd
/usr/bin/pdksh
/usr/bin/bash
/usr/bin/tcsh
/usr/bin/zsh
/usr/bin/fish
/usr/bin/scsh
/usr/bin/rash
/snap/powershell/261/opt/powershell/pwsh

```

</blockQuote>

**Comment by rhubarb-geek-nz** [comment]
> I don't know, @rhubarb-geek-nz. After all, just adding it to `/etc/shells` doesn't mean that it's immediately the login shell, right?

No, but it should mean that `chsh` should not complain so you can change a specific user

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PowerShell/PowerShell/issues/20469#issuecomment-1892885515

Yeah. That works, @rhubarb-geek-nz. Thanks. We probably should have discussed this at https://github.com/PowerShell/PowerShell-Snap/issues/101#issue-2082772353.
